secure/storage/vault/src/lib.rs

```diff
@@ -325,15 +325,30 @@ impl Client {
     }
 
     /// Create or update a key/value pair in a given secret store.
-    pub fn write_secret(&self, secret: &str, key: &str, value: &Value) -> Result<(), Error> {
+    pub fn write_secret(
+        &self,
+        secret: &str,
+        key: &str,
+        value: &Value,
+        version: Option<u32>,
+    ) -> Result<u32, Error> {
+        let payload = if let Some(version) = version {
+            json!({ "data": { key: value }, "options": {"cas": version} })
+        } else {
+            json!({ "data": { key: value } })
+        };
+
         let request = self
             .agent
             .put(&format!("{}/v1/secret/data/{}", self.host, secret));
-        let resp = self
-            .upgrade_request(request)
-            .send_json(json!({ "data": { key: value } }));
+        let resp = self.upgrade_request(request).send_json(payload);
 
-        process_generic_response(resp)
+        if resp.ok() {
```

Comment:
nit: the only downside to this is that we should probably add a fuzzer for the response handling here. But overall it's not a big deal -- I can address it once we land.

Reply:
no more fuzzers here, they look ugly and make the code unmaintainable